server/tests/steps/test_waterfall.py

```diff
@@ -153,3 +153,58 @@ def test_bug_duplicate_rows():
     )
 
     assert_dataframes_equals(result_df, expected_df)
+
+
+def test_waterfall_bug_drill():
+    base_df = pd.DataFrame(
+        {
+            'grand parent': ['A', '1/A', 'i/1/A'] * 2,
+            'parent': ['1/A', 'i/1/A', 'U/i/1/A'] * 2,
+            'label': ['i/1/A', 'U/i/1/A', 'x/U/i/1/A'] * 2,
+            'variable': ['A'] * 3 + ['B'] * 3,
+            'value': [1, 2, 3, 11, 12, 13],
+        }
+    )
+    print(base_df)
+    waterfall_step = WaterfallStep(
+        name='waterfall',
+        valueColumn='value',
+        milestonesColumn='variable',
+        start='A',
+        end='B',
+        labelsColumn='label',
+        parentsColumn='parent',
+        groupby=['grand parent'],
```

Comment:
I'm quite surprise here, I would have guessed that it was the "variable" column which was the group by.
I think then that it would be a clearer example to not use something in the hierarchy as group by column, but an other unrelated dimenion.
Like `type: ["alpha", "beta"]`